packages/@aws-cdk/aws-cognito/lib/user-pool-email.ts

```diff
@@ -172,7 +179,7 @@ class SESEmail extends UserPoolEmail {
       sourceArn: Stack.of(scope).formatArn({
         service: 'ses',
         resource: 'identity',
-        resourceName: encodeAndTest(this.options.fromEmail),
+        resourceName: encodeAndTest(this.options.sesVerifiedDomain ?? this.options.fromEmail),
```

Comment:
If the email address has a different domain than the `sesVerifiedDomain`, a deploy time error occurs:

```
UPDATE_FAILED  | AWS::Cognito::UserPool | myuserpool01998219
Unable to send email message, please try again (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: InvalidParameterException; Request ID: dad6a465-2345-4d13-b1d1-45a038ba1355; Proxy: null
)
```

we need to catch this here, and throw an error if the domain doesn't match (need a unit test for this as well).

Reply:
Done!